### PR TITLE
feat(content): Carbon fiber vehicle frames, swords, advanced 3d lab printer

### PR DESCRIPTION
**Pull request by yay855**
Added in carbon fiber and plastic vehicle frames, carbon fiber and plastic bicycle vehicles, and some more carbon fiber swords to fill out the roster. Renamed carbon fiber sword to carbon fiber katana. Made labs only spawn advanced 3D printers instead of basic ones.

### Required

- [x] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [x] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [ ] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [x] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

## Purpose of change

The 3D printer still has not enough uses to match real-life uses of carbon fiber and plastic materials.

## Describe the solution

Added plastic and carbon fiber vehicle frames and recipes. Added plastic and carbon fiber bicycles, a thing IRL. Changed carbon fiber sword to carbon fiber katana. Added carbon fiber (broad)sword, carbon fiber greatsword, carbon fiber nodachi. Gave advanced 3d printers a chance to spawn in labs instead of basic ones, since the basic ones can be found in electronics stores. Lowered chip resist of carbon fiber material to reflect its more brittle nature. Added alternative recipes for carbon fiber swords to use 'blade' item.

### CODE REVIEW

**Comment by RoyalFox2140** [comment]
Nippon Carbon, folded over 10,000 times.

**Comment by RoyalFox2140** [comment]
Can we check that the vehicle parts will looks_like other parts? I don't see any looks-like json on the vehicle parts.

**Comment by RoyalFox2140** [comment]
So far It looks good, @chaosvolt Needs to review this. Kheir is going to audit the entire 3D printing and carbon fiber lineup of PR's in the distant future as well.

**Review by RoyalFox2140 — APPROVED** [review]
I own a lawn mower sword for home defense because that's what the founding fathers intended.

Obligatory Nippon steel joke was already made.

File: data/json/vehicle_groups.json [hunk 1]
```diff
@@ -16,7 +16,9 @@
       [ "beetle", 500 ],
       [ "bicycle", 800 ],
       [ "bicycle_dirt", 200 ],
+      [ "bicycle_plastic", 400 ],
```
Comment (chaosvolt) [inline data/json/vehicle_groups.json:19]:
```suggestion
      [ "bicycle_plastic", 100 ],
```

File: data/json/vehicleparts/frames.json [hunk 2]
```diff
@@ -135,6 +135,48 @@
...
+    "durability": 500,
+    "description": "A plastic framework.  Other vehicle components can be mounted on it, and it can be attached to other frames to increase the vehicle's size.",
+    "breaks_into": "ig_vp_plastic",
+    "requirements": {
+      "install": { "skills": [ [ "mechanics", 1 ] ], "time": "20 m", "using": [ [ "welding_standard", 10 ] ] },
+      "removal": { "skills": [ [ "mechanics", 1 ] ], "time": "10 m", "using": "vehicle_weld_removal" },
+      "repair": {
+        "skills": [ [ "mechanics", 1 ] ],
+        "time": "5 m",
+        "using": [ [ "welding_standard", 5 ], [ "vehicle_repair_plastic", 1 ] ]
```
Comment (chaosvolt) [inline data/json/vehicleparts/frames.json:153]:
```suggestion
        "using": [ [ "soldering_standard", 5 ], [ "plastics", 1 ] ]
```

File: data/json/requirements/materials.json [hunk 3]
```diff
@@ -282,6 +282,18 @@
     "//": "Materials used for repairing superalloy parts",
     "components": [ [ [ "alloy_sheet", 1 ] ] ]
   },
+  {
+    "id": "vehicle_repair_plastic",
+    "type": "requirement",
+    "//": "Materials used for repairing superalloy parts",
+    "components": [ [ [ "plastic_chunk", 1 ] ] ]
+  },
```
Comment (chaosvolt) [inline data/json/requirements/materials.json:290]: A requirement for using plastic chunks in vehicle repairs already exists.
```suggestion
```

File: data/json/requirements/materials.json [hunk 4]
```diff
@@ -282,6 +282,18 @@
...
+  {
+    "id": "vehicle_repair_carbon",
+    "type": "requirement",
+    "//": "Materials used for repairing superalloy parts",
```
Comment (chaosvolt) [inline data/json/requirements/materials.json:294]:
```suggestion
    "//": "Materials used for repairing carbon fiber parts",
```

File: data/json/recipes/other/vehicles.json [hunk 5]
```diff
@@ -1505,5 +1505,41 @@
...
+  {
+    "result": "frame_plastic",
+    "type": "recipe",
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_VEHICLE",
+    "skill_used": "computer",
+    "difficulty": 1,
+    "autolearn": true,
+    "time": "20 m",
+    "using": [ [ "3d_printing_standard", 10 ] ],
```
Comment (chaosvolt) [inline data/json/recipes/other/vehicles.json:1518]: Number matches the amount of plastic chunks used.
```suggestion
    "using": [ [ "3d_printing_standard", 20 ] ],
```

File: data/json/recipes/weapon/cutting.json [hunk 6]
```diff
@@ -705,6 +705,79 @@
...
+    "result": "sword_carbon",
+    "type": "recipe",
+    "category": "CC_WEAPON",
+    "subcategory": "CSC_WEAPON_CUTTING",
+    "id_suffix": "blade",
+    "skill_used": "computer",
+    "difficulty": 3,
+    "skills_required": [ [ "fabrication", 2 ] ],
+    "//": "Increased time since some forging incorporated into material for the edge",
+    "time": "60 m",
```
Comment (chaosvolt) [inline data/json/recipes/weapon/cutting.json:718]:
```suggestion
    "time": "10 m",
```

File: data/json/recipes/weapon/cutting.json [hunk 7]
```diff
@@ -705,6 +705,79 @@
...
+    "result": "broadsword_carbon",
+    "type": "recipe",
+    "category": "CC_WEAPON",
+    "subcategory": "CSC_WEAPON_CUTTING",
+    "id_suffix": "blade",
+    "skill_used": "computer",
+    "difficulty": 3,
+    "skills_required": [ [ "fabrication", 2 ] ],
+    "//": "Increased time since some forging incorporated into material for the edge",
+    "time": "60 m",
```
Comment (chaosvolt) [inline data/json/recipes/weapon/cutting.json:747]:
```suggestion
    "time": "10 m",
```

File: data/json/recipes/weapon/cutting.json [hunk 8]
```diff
@@ -705,6 +705,79 @@
...
+    "result": "greatsword_carbon",
+    "type": "recipe",
+    "category": "CC_WEAPON",
+    "subcategory": "CSC_WEAPON_CUTTING",
+    "id_suffix": "blade",
+    "skill_used": "computer",
+    "difficulty": 4,
+    "skills_required": [ [ "fabrication", 2 ] ],
+    "//": "Increased time since some forging incorporated into material for the edge",
+    "time": "90 m",
```
Comment (chaosvolt) [inline data/json/recipes/weapon/cutting.json:776]:
```suggestion
    "time": "20 m",
```

File: data/json/recipes/weapon/cutting.json [hunk 9]
```diff
@@ -705,6 +705,79 @@
...
+    "book_learn": [ [ "textbook_computer", 3 ], [ "advanced_electronics", 3 ], [ "reference_fabrication1", 3 ] ],
+    "using": [ [ "3d_printing_standard", 10 ] ],
```
Comment (chaosvolt) [inline data/json/recipes/weapon/cutting.json:778]:
```suggestion
    "using": [ [ "3d_printing_standard", 10 ] ],
```

**Review by chaosvolt — CHANGES_REQUESTED** [review]
Okay so I started with some things I spotted but, before I continue, what weight do you actually want plastic and carbon fiber vehicle frames to be? Right now they're both 4 kg, I assume that's a bit on the heavy side for at least one of those and you would expect the CF frame to be lighter.

I'll look at this more after that's either answered or addressed, more recipes will need their time, printer use, and/or material use set to actually match the items being made because they aren't consistent with the standardization that's since gone into the existing recipes.

Also there's no recipe for the carbon fiber nodachi.

**Comment by RoyalFox2140** [comment]
10 minutes and so on on these forging recipes seems a little fast given they call for blacksmithing standard, thats why I didn't touch it.

**Comment by chaosvolt** [comment]
The ones I had drop the comment and reduce the time are the variant recipes they added, where the blacksmithing and use of steel is swapped out for an already-made blade. Which might warrant 2 in the case of the nodahci but I'd also want to check whether its standard recipe deserves 3 steel_standard like it currently uses or if it would make more sense to use 4 to be double that steel demand of the katana.

**Comment by yay855** [comment]
> Okay so I started with some things I spotted but, before I continue, what weight do you actually want plastic and carbon fiber vehicle frames to be? Right now they're both 4 kg, I assume that's a bit on the heavy side for at least one of those and you would expect the CF frame to be lighter.
> 
> I'll look at this more after that's either answered or addressed, more recipes will need their time, printer use, and/or material use set to actually match the items being made because they aren't consistent with the standardization that's since gone into the existing recipes.
> 
> Also there's no recipe for the carbon fiber nodachi.

I reduced the weight of the carbon fiber and plastic frames, you're right that they'd be even lighter than wood ones. I also added recipes for the carbon fiber nodachi, can't believe I forgot that.

**Comment by chaosvolt** [comment]
Nice, will look it over in a bit.

File: data/json/items/vehicle/frames.json [hunk 10]
```diff
@@ -96,5 +96,29 @@
...
+    "price": "35 USD",
+    "price_postapoc": "1 USD",
+    "copy-from": "foldframe"
+  },
+  {
+    "type": "GENERIC",
+    "id": "frame_carbon",
+    "name": { "str": "carbon fiber frame" },
+    "description": "A large frame made of plastic.  Useful for lightweight yet sturdy vehicles.",
+    "weight": "2500 g",
```
Comment (chaosvolt) [inline data/json/items/vehicle/frames.json:117]: Mebbe just a lil lower so it's lighter than plastic, like the difference between plastic and carbon fiber boat hulls. Went with the common 0.6x for the suggestion, contrast fiber fiber boat hulls being a third the weight of plastic ones.
```suggestion
    "weight": "1500 g",
```

File: data/json/recipes/other/vehicles.json [hunk 11]
```diff
@@ -1505,5 +1505,41 @@
     "book_learn": [ [ "textbook_computer", 1 ], [ "advanced_electronics", 1 ], [ "reference_fabrication1", 1 ] ],
     "using": [ [ "3d_printing_standard", 10 ] ],
     "components": [ [ [ "carbon_fiber_filament", 100 ] ] ]
+  },
+  {
+    "result": "frame_plastic",
+    "type": "recipe",
```
Comment (chaosvolt) [inline data/json/recipes/other/vehicles.json:1511]:
```suggestion
    "type": "recipe",
    "id_suffix": "printed",
```

File: data/json/recipes/other/vehicles.json [hunk 12]
```diff
@@ -1505,5 +1505,41 @@
...
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_VEHICLE",
+    "skill_used": "computer",
+    "difficulty": 1,
+    "autolearn": true,
+    "time": "20 m",
+    "using": [ [ "3d_printing_standard", 20 ] ],
+    "components": [ [ [ "plastic_chunk", 20 ] ] ]
```
Comment (chaosvolt) [inline data/json/recipes/other/vehicles.json:1519]: 2500 divided by 50
```suggestion
    "time": "50 m",
    "using": [ [ "3d_printing_standard", 50 ] ],
    "components": [ [ [ "plastic_chunk", 50 ] ] ]
```

File: data/json/recipes/other/vehicles.json [hunk 13]
```diff
@@ -1505,5 +1505,41 @@
...
+    "type": "recipe",
+    "result": "frame_plastic",
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_VEHICLE",
+    "skill_used": "fabrication",
+    "difficulty": 3,
+    "time": "40 m",
+    "autolearn": true,
+    "tools": [ [ [ "mold_plastic", -1 ] ], [ [ "surface_heat", 30, "LIST" ] ] ],
+    "components": [ [ [ "plastic_chunk", 20 ] ] ]
```
Comment (chaosvolt) [inline data/json/recipes/other/vehicles.json:1531]: Compare standard plastic boat hull using 50 doses of `surface_heat` and taking 1.5 times longer than the comparable printer recipe.
```suggestion
    "time": "75 m",
    "autolearn": true,
    "tools": [ [ [ "mold_plastic", -1 ] ], [ [ "surface_heat", 75, "LIST" ] ] ],
    "components": [ [ [ "plastic_chunk", 50 ] ] ]
```

File: data/json/recipes/other/vehicles.json [hunk 14]
```diff
@@ -1505,5 +1505,41 @@
...
+    "result": "frame_carbon",
+    "type": "recipe",
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_VEHICLE",
+    "skill_used": "computer",
+    "difficulty": 2,
+    "autolearn": true,
+    "time": "20 m",
+    "using": [ [ "3d_printing_standard", 10 ] ],
+    "components": [ [ [ "carbon_fiber_filament", 200 ] ] ]
```
Comment (chaosvolt) [inline data/json/recipes/other/vehicles.json:1543]: If one goes for the 1500 grams suggestion, 1 unit of filament being 5 grams each
```suggestion
    "time": "30 m",
    "using": [ [ "3d_printing_standard", 30 ] ],
    "components": [ [ [ "carbon_fiber_filament", 300 ] ] ]
```

File: data/json/recipes/weapon/cutting.json [hunk 15]
```diff
@@ -705,6 +705,104 @@
...
+    "type": "recipe",
+    "category": "CC_WEAPON",
+    "subcategory": "CSC_WEAPON_CUTTING",
+    "skill_used": "computer",
+    "difficulty": 4,
+    "skills_required": [ [ "fabrication", 2 ] ],
+    "//": "Increased time since some forging incorporated into material for the edge",
+    "time": "90 m",
+    "book_learn": [ [ "textbook_computer", 3 ], [ "advanced_electronics", 3 ], [ "reference_fabrication1", 3 ] ],
+    "using": [ [ "3d_printing_standard", 10 ], [ "steel_standard", 3 ], [ "blacksmithing_standard", 12 ] ],
```
Comment (chaosvolt) [inline data/json/recipes/weapon/cutting.json:761]:
```suggestion
    "using": [ [ "3d_printing_standard", 20 ], [ "steel_standard", 3 ], [ "blacksmithing_standard", 12 ] ],
```

File: data/json/recipes/weapon/cutting.json [hunk 16]
```diff
@@ -705,6 +705,104 @@
...
+    "type": "recipe",
+    "category": "CC_WEAPON",
+    "subcategory": "CSC_WEAPON_CUTTING",
+    "id_suffix": "blade",
+    "skill_used": "computer",
+    "difficulty": 4,
+    "skills_required": [ [ "fabrication", 2 ] ],
+    "time": "20 m",
+    "book_learn": [ [ "textbook_computer", 3 ], [ "advanced_electronics", 3 ], [ "reference_fabrication1", 3 ] ],
+    "using": [ [ "3d_printing_standard", 10 ] ],
```
Comment (chaosvolt) [inline data/json/recipes/weapon/cutting.json:775]:
```suggestion
    "using": [ [ "3d_printing_standard", 20 ] ],
```

File: data/json/recipes/weapon/cutting.json [hunk 17]
```diff
@@ -705,6 +705,104 @@
...
+    "type": "recipe",
+    "category": "CC_WEAPON",
+    "subcategory": "CSC_WEAPON_CUTTING",
+    "skill_used": "computer",
+    "difficulty": 4,
+    "skills_required": [ [ "fabrication", 2 ] ],
+    "//": "Increased time since some forging incorporated into material for the edge",
+    "time": "90 m",
+    "book_learn": [ [ "textbook_computer", 3 ], [ "advanced_electronics", 3 ], [ "reference_fabrication1", 3 ] ],
+    "using": [ [ "3d_printing_standard", 10 ], [ "steel_standard", 3 ], [ "blacksmithing_standard", 12 ] ],
```
Comment (chaosvolt) [inline data/json/recipes/weapon/cutting.json:789]:
```suggestion
    "using": [ [ "3d_printing_standard", 20 ], [ "steel_standard", 3 ], [ "blacksmithing_standard", 12 ] ],
```

File: data/json/recipes/weapon/cutting.json [hunk 18]
```diff
@@ -705,6 +705,104 @@
...
+    "type": "recipe",
+    "category": "CC_WEAPON",
+    "subcategory": "CSC_WEAPON_CUTTING",
+    "id_suffix": "blade",
+    "skill_used": "computer",
+    "difficulty": 4,
+    "skills_required": [ [ "fabrication", 2 ] ],
+    "time": "20 m",
+    "book_learn": [ [ "textbook_computer", 3 ], [ "advanced_electronics", 3 ], [ "reference_fabrication1", 3 ] ],
+    "using": [ [ "3d_printing_standard", 10 ] ],
```
Comment (chaosvolt) [inline data/json/recipes/weapon/cutting.json:803]:
```suggestion
    "using": [ [ "3d_printing_standard", 20 ] ],
```

**Review by chaosvolt — CHANGES_REQUESTED** [review]
I think that should cover it.

**Review by chaosvolt — APPROVED** [review]

**Pull request merged**